src/Sylius/Bundle/CoreBundle/Grid/Provider/ConfigurableProvider.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Grid\Provider;
+
+use Psr\Container\ContainerInterface;
+use Sylius\Component\Grid\Definition\Grid;
+use Sylius\Component\Grid\Provider\GridProviderInterface;
+use Webmozart\Assert\Assert;
+
+final class ConfigurableProvider implements GridProviderInterface
+{
+    /** @param array<string, mixed> $configuration */
+    public function __construct(
+        private ContainerInterface $providers,
+        private array $configuration,
+    ) {
+    }
+
+    public function get(string $code): Grid
+    {
+        /** @var string|null $type */
+        $type = $this->configuration['grids'][$code]['type'] ?? null;
+
+        if (null === $type) {
+            return $this->getDefaultProvider()->get($code);
```

Comment:
Since this function accessing `$this->configuration` directly, I feel like `getDefaultProvider` function is unnecessary.

```php
if (null === $type) {
    $type = $this->configuration['default_type'] ?? null;
    Assert::notNull($type, 'Default grid provider type must be set.'):
}
```